agent/websockets/shim.go

```diff
@@ -225,6 +229,7 @@ func shimBody(resp *http.Response, shimCode string) error {
 		reader: io.MultiReader(strings.NewReader(prefix), wrapped),
 		closer: wrapped,
 	}
+	resp.Header.Del("Content-Length")
```

Comment:
Is this to address the content-length mismatch error you mentioned?  If so, then the nit about separating bug fixes for the websocket issue from this batching feature would apply here too

Reply:
Yes; and again, I should have separated it into its own PR. I will be careful about that in the future.